java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S3077_java.json

```diff
@@ -0,0 +1,20 @@
+{
+  "title": "Non-primitive fields should not be \"volatile\"",
+  "type": "BUG",
+  "status": "ready",
+  "remediation": {
+    "func": "Constant\/Issue",
+    "constantCost": "20min"
+  },
+  "tags": [
+    "multi-threading",
+    "cert"
+  ],
+  "standards": [
+    "CERT"
+  ],
+  "defaultSeverity": "Major",
```

Comment:
@benzonico any thought about being major bug?

Reply:
The "bug" might be arguable given the amount of issues in ruling (and in jdk/guava especially)